packages/tools/devtools/devtools-core/src/CommonInterfaces.ts

```diff
@@ -45,3 +48,41 @@ export interface HasFluidObjectId {
 	 */
 	fluidObjectId: FluidObjectId;
 }
+
+/**
+ * Base interface used in message data for edits allowed by a particular Fluid object (DDS) via
+ * an enum.
+ *
+ * @public
+ */
+export enum EditType {
+	number = "number",
+	string = "string",
+	boolean = "boolean",
+}
+
+/**
+ * Base interface used in message data for edit events containing the allowed edits as an array of EditType.
+ *
+ * @internal
+ */
+export interface HasEditType {
+	editType: EditType;
+}
+
+/**
+ * Base interface used in message data for communicating edits
+ * @public
```

Comment:
We should be able to make this `@internal` after we remove the editors from `ContainerDevtoolsProps`.

Reply:
The only one I was not able to make internal was enum EditType in commonInterfaces.ts since it is used by editProps, the new field in VisualNodeBase. Any way aorund this or is this fine?

Reply:
I have a PR up (#16569) that removes data visualization override capabilities from the API surface (for now). Once that goes in, you should be able to make the remaining items internal, and will likely be able to remove some of the new editing-related types from the package-root exports (src/index.ts) altogether.

Reply:
That PR has been merged. So if you merge / rebase onto main, you should be able to make this internal :)